sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedIterable.java

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.rest;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *  This class provides utility to iterate over {@link PagedResponse}.
+ * @param  <T> value
+ */
+public class PagedIterable<T> implements Iterable<T> {
+    private final PagedFlux<T> pagedFlux;
+
+    /**
+     * Creates instance given {@link Flux}.
+     * @param flux  the flux value
+     */
+    public PagedIterable(Flux<T> flux) {
+        this.pagedFlux = new PagedFlux<>(getSupplier(flux));
```

Comment:
If this is a single-paged response, is there a need for `PagedIterable`? Should the `PagedIterable` be constructor only through `PagedFlux`?